client/qiskit_serverless/core/clients/serverless_client.py

```diff
@@ -177,6 +177,16 @@ def _verify_credentials(self):
                 "Credentials couldn't be verified."
             ) from reason
 
+    def dependencies_versions(self):
+        """Get the allowed additional dependencies and versions."""
```

Comment:
Similar to other comments, we should unify how we want to describe the functionality, `additional`, `dynamic`, `available`...